lib/cretonne/meta/cdsl/typevar.py

```diff
@@ -331,6 +340,67 @@ def double_vector(self):
 
         return new
 
+    def map(self, func):
+        # type: (str) -> TypeSet
+        """
+        Return the image of self across the derived function func
+        """
+        if (func == TypeVar.SAMEAS):
+            return self
+        elif (func == TypeVar.LANEOF):
+            return self.lane_of()
+        elif (func == TypeVar.ASBOOL):
+            return self.as_bool()
+        elif (func == TypeVar.HALFWIDTH):
+            return self.half_width()
+        elif (func == TypeVar.DOUBLEWIDTH):
+            return self.double_width()
+        elif (func == TypeVar.HALFVECTOR):
+            return self.half_vector()
+        elif (func == TypeVar.DOUBLEVECTOR):
+            return self.double_vector()
+        else:
+            assert False, "Unknown derived function: " + func
+
+    def map_inverse(self, func):
+        # type: (str) -> TypeSet
+        """
+        Return the inverse image of self across the derived function func
```

Comment:
I think I would prefer to cal these two methods `image` and `preimage`. See http://mathworld.wolfram.com/Preimage.html